src/api/component.js

```diff
@@ -115,19 +116,21 @@ Component.extend = function extend(definition = {}, Base = this) {
 // Skate
 //
 // This is a default implementation that does strict equality copmarison on
-// prevoius props and next props. It synchronously renders on the first prop
+// previous props and next props. It synchronously renders on the first prop
 // that is different and returns immediately.
 Component.updated = function updated(elem, prev) {
   if (!prev) {
     return true;
   }
-
-  // eslint-disable-next-line no-restricted-syntax
-  for (const name in prev) {
-    if (prev[name] !== elem[name]) {
+  // use get all keys so that we check Symbols as well as regular props
+  // using a for loop so we can break early
```

Comment:
In some perf testing I've done, `for` is also faster in performance critical code than `Array` methods. Especially in older browsers.